### PR TITLE
TSCLibc: add an implementation of `mkdtemp` for Windows

### PR DESCRIPTION
**Pull request by compnerd**
This adds an implementation of `mkdtemp` for Windows to accommodate
swift-package-manager which uses this rather than Foundation.

### CODE REVIEW

**Comment by compnerd** [comment]
CC: @tomerd @neonichu @abertelrud

**Comment by compnerd** [comment]
@swift-ci please test

File: Sources/TSCLibc/libc.swift [hunk 1]
```diff
@@ -29,10 +29,45 @@ public func realpath(
 }
 
 // char *mkdtemp(char *template);
+// NOTE(compnerd) this is unsafe!  This assumes that the template is *ASCII*.
 public func mkdtemp(
     _ template: UnsafeMutablePointer<CChar>?
 ) -> UnsafeMutablePointer<CChar>? {
-  fatalError("mkdtemp is unimplemented")
```
Comment (tomerd) [inline Sources/TSCLibc/libc.swift:35]: was this just not used before?

Comment (compnerd) [reply Sources/TSCLibc/libc.swift:35]: It "was" in the sense that s-p-m never got to the point of building stuff previously.  I've had this change laying around for ages.  Just draining changes that will allow s-p-m to get further :)

Comment (tomerd) [reply Sources/TSCLibc/libc.swift:35]: ah gotcha

**Comment by tomerd** [comment]
@compnerd please merge when ready

**Pull request merged**